pkg/destroy/bootstrap/bootstrap.go

```diff
@@ -38,10 +38,10 @@ func Destroy(dir string) (err error) {
 	tfPlatformVarsFileName := fmt.Sprintf(cluster.TfPlatformVarsFileName, platform)
 
 	// Azure Stack uses the Azure platform but has its own Terraform configuration.
-	// Set platform to azurestack after setting tfPlatformVarsFileName, because the
+	// Set platform to AzureStackCloud after setting tfPlatformVarsFileName, because the
 	// tfvars file is still terraform.azure.auto.tfvars.json in the case of Azure Stack.
 	if platform == typesazure.Name && metadata.Azure.CloudName == typesazure.StackCloud {
-		platform = "azurestack"
+		platform = typesazure.StackCloud.Name()
```

Comment:
I don't think that this is the best solution. We put the tf files under data/data/azurestack as so we call the platform "azurestack" in the stages.
https://github.com/openshift/installer/blob/ec1c83027c3963abb273f412c60472db58bf1860/pkg/terraform/stages/azure/stages.go#L20-L22
So I think we should call the platform "azurestack" here as well, for consistency. We should instead the following line to "azurestack".
https://github.com/openshift/installer/blob/ec1c83027c3963abb273f412c60472db58bf1860/pkg/terraform/stages/platform/stages.go#L31

Reply:
@staebler, your point makes sense. I ended up going a bit further and eliminated the use of the "azurestack" magic string. IMHO, it is better to nip this in the bud and clean it up now (mess I introduced); but if you think this is overkill let me know.